x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -615,6 +620,23 @@ public void checkPrivileges(
             listener = originalListener;
         }
 
+        // could also check role has application privilege here
+        if (applicationPrivileges.isEmpty() || false == Transports.isTransportThread(Thread.currentThread())) {
```

Comment:
This is not quite the right check yet; we want to only run this check if the role has (many) application privileges and the request is checking for them.